### PR TITLE
CE-2677 Make the destroying session data method testable and write the tests

### PR DESCRIPTION
**Pull request by adamkarminski**
/cc @kamilkoterba @Grunny

### CODE REVIEW

**Comment by kamilkoterba** [comment]
LGTM

File: extensions/wikia/ContentReview/tests/ContentReviewHooksTest.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+	/**
+	 * In this test case the setSessionData() method should not be called
+	 * because there are no IDs of wikias stored in a session under the test mode key.
+	 */
+	public function testDisableTestModeNoSessionData() {
+		$requestMock = $this->getMockBuilder( 'WebRequest' )
+			->disableOriginalConstructor()
+			->setMethods( [ 'getSessionData', 'setSessionData' ] )
+			->getMock();
```
Comment (Grunny) [inline extensions/wikia/ContentReview/tests/ContentReviewHooksTest.php]: It's simpler to just use `getMock` unless you need to disable the constructor, which you don't here, i.e. just:
`$requestMock = $this->getMock( 'WebRequest', [ 'getSessionData', 'setSessionData' ] );`

**Comment by Grunny** [comment]
I think it would be better to just fully test the hook, i.e. test `onUserLogoutComplete` by simply injecting a mocked `User` object and just mock its `getRequest` method. I'm not sure of the value of splitting out a method and only testing that here. Then `disableTestMode` should be made private as it's an implementation detail (and either static, or switch the entire class to have the hooks be non-static using one static method with the `Hooks::register` calls), or it could even end up in a helper class if there is enough need.

**Comment by kamilkoterba** [comment]
Great!

**Pull request merged**